civictechprojects/views.py

```diff
@@ -143,21 +125,13 @@ def projects_list(request):
         url_parts = request.GET.urlencode()
         query_params = urlparse.parse_qs(
             url_parts, keep_blank_values=0, strict_parsing=0)
-        projects = Project.objects
-        if 'keyword' in query_params:
-            projects = (
-                projects
-                .filter(
-                    project_description__icontains=query_params['keyword'][0],
-                )
-            )
-        if 'issueArea' in query_params:
-            projects = (
-                projects
-                .filter(
-                    project_issue_area__name__in=query_params['issueArea'],
-                )
-            )
+        projects = (
```

Comment:
I see what you're saying about complicated query logic. I think this is partly because Python handles ternaries as

`resultIfTrue if cond else resultIfFalse`

rather than

`cond ? resultIfTrue : resultIfFalse`

which I agree is a little weird to grok.

However, I don't think the query logic can be refactored as suggested. The current form of the query construction is

`filtered_results if filters_exist else all_results`

so as to deal with the case of an empty result set. If filters were passed to the API and the filtered result set is empty, then we should return an empty result set. However, if the result set is empty because no filters were passed, then we should return a result containing all possible projects.

Lemme know if I'm not making sense.

Reply:
Ah, right, we need to chain filters in this case so you could do something like:
```
projects = Project.objects
if x:
  projects = projects.filter_by_x(x)
if y:
  projects = projects.filter_by_y(x)
```